components/client/src/test/unit/java/com/hotels/styx/client/StyxHttpClientTest.java

```diff
@@ -180,470 +107,360 @@ private static BackendService backendWithOrigins(int originPort) {
-            server.stubFor(WireMock.get(urlStartingWith("/")).willReturn(aResponse().withStatus(200)));
-            portConsumer.accept(port);
-        } finally {
-            server.stop();
-        }
+    private Connection mockConnection(Origin origin, Observable<HttpResponse> response) {
+        Connection connection = mock(Connection.class);
+        when(connection.write(any(HttpRequest.class))).thenReturn(response);
+        when(connection.getOrigin()).thenReturn(SOME_ORIGIN);
+        return connection;
     }
 
-    @Test
-    public void asksTheConnectionPoolForAConnection() {
-        Connection connection = aConnection();
-        ConnectionPool pool = mockPool(connection);
-        StyxHttpClient styxHttpClient = newStyxHttpClient(pool);
-        styxHttpClient.sendRequest(SOME_REQ).subscribe(DO_NOTHING);
-        verify(pool).borrowConnection();
+    private LoadBalancingStrategy mockLbStrategy(ConnectionPool pool) {
+        LoadBalancingStrategy lbStrategy = mock(LoadBalancingStrategy.class);
+        when(lbStrategy.vote(any(LoadBalancingStrategy.Context.class))).thenReturn(ImmutableList.of(remoteHost(pool.getOrigin(), pool, mock(StyxHostHttpClient.class))));
+        return lbStrategy;
     }
 
     @Test
-    public void releasesConnectionWhenOperationExecutionIsFinished() {
-        Connection connection = aConnection();
-        ConnectionPool pool = mockPool(connection);
-        StyxHttpClient styxHttpClient = newStyxHttpClient(pool);
-        TestSubscriber<HttpResponse> subscriber = new TestSubscriber<>();
-        styxHttpClient.sendRequest(SOME_REQ).subscribe(subscriber);
-        subscriber.awaitTerminalEvent();
-        verify(pool).returnConnection(connection);
-    }
-
-    @Test
-    public void closesConnectionWhenTransactionIsUnsubscribed() throws IOException {
-        PublishSubject<HttpResponse> responseSubject = PublishSubject.create();
-        requestOperationFactory = newRequestOperationFactory(responseSubject);
-
-        Connection connection = aConnection();
-        ConnectionPool pool = mockPool(connection);
-        BackendService backendService = backendWithOrigins(connection.getOrigin().host().getPort());
-
-        OriginsInventory originsInventory = newOriginsInventoryBuilder(backendService.id())
-                .connectionPoolFactory(origin -> pool)
-                .initialOrigins(backendService.origins())
-                .build();
+    public void sendsRequestToHostChosenByLoadBalancer() {
+        StyxHostHttpClient hostClient = mockHostClient(just(response(OK).build()));
 
         StyxHttpClient styxHttpClient = new StyxHttpClient.Builder(backendService)
-                .originsInventory(originsInventory)
+                .metricsRegistry(metricRegistry)
+                .loadBalancingStrategy(
+                        mockLoadBalancer(
+                                asList(new RemoteHostWrapper(SOME_ORIGIN.id(), SOME_ORIGIN, mock(ConnectionPool.class), hostClient))
```

Comment:
There are several instances when `asList` is used unnecessarily, when `Collections.emptyList` or `Collections.singletonList` could be used.

Reply:
I am aware that we could have used `emptyList` or `singletonList` in place of `asList`.

There is no reason to use 3 different methods just to create a list. Why should I use a different method to create a list simply because it is empty, or simply because it has just one element? 

I want consistency over clutter in this test.

Reply:
I believe that the choice of method can make the intentions of the test more obvious to readers, but this is a pretty minor concern, so it's fine to just leave it as it is.